lib/ui/painting/image_encoding.cc

```diff
@@ -41,77 +41,67 @@ void InvokeDataCallback(std::unique_ptr<DartPersistentValue> callback,
   }
 }
 
-sk_sp<SkData> EncodeImage(sk_sp<SkImage> image,
-                          SkEncodedImageFormat format,
-                          int quality) {
+sk_sp<SkData> GetImageBytesAsRGBA(sk_sp<SkImage> image) {
   if (image == nullptr) {
     return nullptr;
   }
-  return image->encodeToData(format, quality);
+
+  // Copy the GPU image snapshot into CPU memory.
+  auto cpu_snapshot = image->makeRasterImage();
+  if (!cpu_snapshot) {
+    FXL_LOG(ERROR) << "Pixel copy failed.";
+    return nullptr;
+  }
+
+  SkPixmap pixmap;
+  if (!cpu_snapshot->peekPixels(&pixmap)) {
+    FXL_LOG(ERROR) << "Pixel address is not available.";
+    return nullptr;
+  }
+
+  if (pixmap.colorType() != kRGBA_8888_SkColorType) {
```

Comment:
@chinmaygarde is this ever possible?

Reply:
Yes. It usually depends on the image (for example, it could be grayscale). We probably want to peek the pixels from the the image and also return a format type to the caller. If you want to make sure all color formats are converted to N32, then you'd need to create a N32 raster surface and perform the conversion there. Though I think that would be wasteful.

Reply:
I updated this to convert the color format because I think it yields the cleanest API.  Let me know what you think.

Reply:
I updated this to convert the color format because I think it yields the cleanest API (and because it caters to the 95% use case, which is images that are already in RGBA).  Let me know what you think.

Reply:
FYI, I've written up the options that I see for the API in [this doc](https://docs.google.com/document/d/1nkl-8Sv5r530N4eFyKDZKeJgwVt83TqdmqdAvtizlCU/view).

This PR currently goes with option 1.  I'm pretty strongly opposed to options 2 & 3.  I actually think option 4 is probably the best choice, but it's also something that we could add on later if we go with option 1.  And since we want to fix the existing API (and the binary size bloat it causes) before the next beta release, I like the idea of going with option 1 and keeping our options open to expand to option 4 at a later time.